src/server/pps/server/master.go

```diff
@@ -180,16 +180,28 @@ func (a *apiServer) setPipelineCrashing(ctx context.Context, pipelineName string
 	return a.setPipelineState(ctx, pipelineName, pps.PipelineState_PIPELINE_CRASHING, reason)
 }
 
-// every running pipeline with standby == true has a corresponding goroutine
+// Every running pipeline with standby == true has a corresponding goroutine
 // running monitorPipeline() that puts the pipeline in and out of standby in
 // response to new output commits appearing in that pipeline's output repo
 func (a *apiServer) cancelMonitor(pipeline string) {
 	a.monitorCancelsMu.Lock()
+	defer a.monitorCancelsMu.Unlock()
 	if cancel, ok := a.monitorCancels[pipeline]; ok {
 		cancel()
 		delete(a.monitorCancels, pipeline)
 	}
-	a.monitorCancelsMu.Unlock()
+}
+
+// Every crashing pipeline has a corresponding goro running
+// monitorCrashingPipeline that checks to see if the issues have resolved
+// themselves and moves the pipeline out of crashing if they have.
+func (a *apiServer) cancelCrashingMonitor(pipeline string) {
```

Comment:
I think this might need to be called in `deletePipelineResources`, so that when a pipeline is updated, leftover `cancelCrashingMonitor` goros get shut down and don't accidentally clobber the state of the updated pipeline